packages/lib/services/plugins/api/JoplinImaging.ts

```diff
@@ -5,16 +5,38 @@ import Setting from '../../../models/Setting';
 import shim from '../../../shim';
 import { Rectangle } from './types';
 
-export interface Implementation {
-	nativeImage: any;
-}
-
 export interface CreateFromBufferOptions {
 	width?: number;
 	height?: number;
 	scaleFactor?: number;
 }
 
+export interface CreateFromPdfOptions {
+	/**
+	 * The first page to export. Defaults to `1`, the first page in
+	 * the document.
+	 */
+	minPage?: number;
+
+	/**
+	 * The number of the last page to convert. Defaults to the last page
+	 * if not given.
+	 *
+	 * If `maxPage` is greater than the number of pages in the PDF, all pages
+	 * in the PDF will be converted to images.
+	 */
+	maxPage?: number;
```

Comment:
Do we need to expose this for now? Since we don't have an API to get this info from the pdf in the first place

Reply:
Because PDF-to-image conversion can be slow, I would prefer to have some way to only load a specific number of pages (similar to the `cursor` we use for the data API). The intention for `minPage` and `maxPage` was to allow fetching only a certain number of pages, starting at a specific point.

An alternative could be to allow users to provide a `cursor` and a `limit` (and also return a `has_more` property), similar to the data API.

Reply:
I think the `cursor` interface would be a bit too abstract for this, so it's fine as it is, however how will the plugin know the number of pages in the document?

Reply:
> however how will the plugin know the number of pages in the document?

It currently isn't possible to get the number of pages in the document. However, it's fine if `maxPage` is greater than the maximum number of pages. As such, a plugin could do something similar to the following:
```ts
for (let i = 0; ; i += 5) {
    const images = await joplin.imaging.createFromPdfResource(
        resourceId,
        { minPage: i, maxPage: i + 5 },
    );
    if (images.length === 0) {
        break;
    }
    await processImages(images);
}
```

It could make sense to add the above to the test plugin (which currently processes only the first 10 pages).

Reply:
It could also make sense to have `joplin.imaging.createFromPdfResource` return an object with additional PDF information, in addition to image handles. For example,
```
{
    pages: Handle[]; // String image handles for each page
    pdfInfo: {
        pageCount: number;
    },
}
```

Such an object would also make it easier to return more information about PDFs in the future, without breaking changes to the plugin API.